libretroshare/src/pqi/p3peermgr.cc

```diff
@@ -1071,42 +1071,53 @@ bool p3PeerMgrIMPL::addFriend(const RsPeerId& input_id, const RsPgpId& input_gpg
 }
 
 
-bool p3PeerMgrIMPL::addSslOnlyFriend( const RsPeerId& sslId, const RsPgpId& pgp_id,const RsPeerDetails& dt )
+bool p3PeerMgrIMPL::addSslOnlyFriend(
+        const RsPeerId& sslId, const RsPgpId& pgp_id, const RsPeerDetails& dt )
 {
-	if(sslId.isNull() || sslId == getOwnId())
-    {
-        RsErr() <<"Attempt to add yourself or a null ID as SSL-only friend (id=" << sslId << ")" << std::endl;
-        return false;
-    }
+	if(sslId.isNull())
+	{
+		RsErr() << __PRETTY_FUNCTION__ << " Cannot add a null "
+		        << "ID as SSL-only friend " << std::endl;
+		return false;
+	}
 
+	if(pgp_id.isNull())
+	{
+		RsErr() << __PRETTY_FUNCTION__ << " Cannot add as SSL-only friend a "
+		        << "peer with null PGP" << std::endl;
+		return false;
+	}
+
+	if(sslId == getOwnId())
+	{
+		RsErr() << __PRETTY_FUNCTION__ << " Cannot add yourself as SSL-only "
+		        << "friend (id=" << sslId << ")" << std::endl;
+		   return false;
+	}
+
+	bool alreadyFriend = false;
 	peerState pstate;
+	pstate.skip_pgp_signature_validation = true;
```

Comment:
Moving that here seems wrong to me: because the flag will be set even if the short invite received is wrong because it has a known (friend) SSL id with the wrong PGP id.

Reply:
In that case `pstate` is completely discarded, while if we already have it the `pstate` and thus `pstate.skip_pgp_signature_validation` is overwritten with the one we have already. That flag would apply only in case is a completely new ssl-only friend

Reply:
also, the way it is now, you can re-add with short invite some friend you already have and it will reset his PGP signature validation. This is not the expected behavior! Really, you dont set such a flag to true *unless* all checks have been passed.

Reply:
no it will not, look at line 1118 `pstate` and thus `pstate.skip_pgp_signature_validation` get overriden with the already known state if it is already friend. So if it has already PGP `pstate.skip_pgp_signature_validation` will be reset to `false`, if not it will remain `true`

Reply:
what is the motivation behind having it up here?

Reply:
my point is: with such a dangerous flag, we need to be as clear as possible. Moving that =true up here requires lots of thinking in order to make sure that it does the correct thing in all cases. Whatever you gain with this is not worth it IMHO.

Reply:
Wait I found another risky and quircky case, I am writing the code to handle that too

Reply:
> what is the motivation behind having it up here?

The motivation was so it get overridden by already known value if it was an already known location, but now it is irrelevant I have just pushed a safer version ;)

I really appreciate @csoler security insight